app/models/managed_interactive.rb

```diff
@@ -0,0 +1,269 @@
+    "managed interactive"
+  end
+
+  def self.portal_type
+    "iframe interactive"
+  end
+
+  def storage_key
+    if name.present?
+      "#{interactive_page.lightweight_activity.id}_#{interactive_page.id}_#{id}_#{self.class.to_s.underscore.gsub(/\//, '_')}_#{name.downcase.gsub(/ /, '_')}"
+    else
+      "#{interactive_page.lightweight_activity.id}_#{interactive_page.id}_#{id}_#{self.class.to_s.underscore.gsub(/\//, '_')}"
+    end
+  end
+
+  def to_hash
+    # Deliberately ignoring user (will be set in duplicate)
+    {
+      library_interactive_id: library_interactive_id,
+      name: name,
+      url_fragment: url_fragment,
+      authored_state: authored_state,
+      is_hidden: is_hidden,
+      is_full_width: is_full_width,
+      show_in_featured_question_report: show_in_featured_question_report,
+      inherit_aspect_ratio_method: inherit_aspect_ratio_method,
+      custom_aspect_ratio_method: custom_aspect_ratio_method,
+      inherit_native_width: inherit_native_width,
+      custom_native_width: custom_native_width,
+      inherit_native_height: inherit_native_height,
+      custom_native_height: custom_native_height,
+      inherit_click_to_play: inherit_click_to_play,
+      custom_click_to_play: custom_click_to_play,
+      inherit_full_window: inherit_full_window,
+      custom_full_window: custom_full_window,
+      inherit_click_to_play_prompt: inherit_click_to_play_prompt,
+      custom_click_to_play_prompt: custom_click_to_play_prompt,
+      inherit_image_url: inherit_image_url,
+      custom_image_url: custom_image_url
+    }
+  end
+
+  def portal_hash
+    iframe_data = to_hash
+    iframe_data[:type] = 'iframe_interactive'
+    iframe_data[:id] = id
+    iframe_data[:display_in_iframe] = reportable_in_iframe?
+    iframe_data
+  end
+
+  def report_service_hash
+    {
+      type: 'iframe_interactive',
+      id: embeddable_id,
+      name: name,
+      url: url,
+      show_in_featured_question_report: show_in_featured_question_report,
+      display_in_iframe: reportable_in_iframe?,
+      width: native_width,
+      height: native_height,
+      question_number: index_in_activity
+    }
+  end
+
+  def duplicate
+    # Generate a new object with those values
+    ManagedInteractive.new(self.to_hash)
+    # N.B. the duplicate hasn't been saved yet
+  end
+
+  def export
+    json = self.as_json(only:[:name,
+      :url_fragment,
+      :authored_state,
+      :is_hidden,
+      :is_full_width,
+      :show_in_featured_question_report,
+      :inherit_aspect_ratio_method,
+      :custom_aspect_ratio_method,
+      :inherit_native_width,
+      :custom_native_width,
+      :inherit_native_height,
+      :custom_native_height,
+      :inherit_click_to_play,
+      :custom_click_to_play,
+      :inherit_full_window,
+      :custom_full_window,
+      :inherit_click_to_play_prompt,
+      :custom_click_to_play_prompt,
+      :inherit_image_url,
+      :custom_image_url
+    ])
+    if library_interactive
```

Comment:
It should always have a library interactive (at least based on code above) so this check could be removed.